### PR TITLE
Added release branch and poetry details to the README

### PR DESCRIPTION
**Pull request by jacobcook1995**
I've added details of what I think the correct publication process with `poetry` is to the README. Let me know if this makes sense, I've only tested the commands up to a certain point, so if anything is wrong there might be changes down the line.

I also bumped the version with `poetry`, as it seemed like a sensible time to do so (though it admitted conflicts with the procedure I set out for bumping it).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,21 +52,64 @@ whenever commits are made to this repository and for all pull requests.
...
+
+This gives all developers an opportunity to fix problems they have noticed, prior to
+release. This can be done by making pull requests against the `release` branch. Once
+these changes have been made, documentation has been checked, and all tests pass, the
+final commit should bump the version using `poetry` so that the correct version is
+recorded in `pyproject.toml`.
+
+```bash
+poetry version x.y.z
+```
```
Comment (davidorme) [inline README.md:77]: Might want to use `poetry bump [major/minor/patch]` here? Or as alternative to `poetry version`

File: README.md [hunk 2]
```diff
@@ -52,21 +52,64 @@ whenever commits are made to this repository and for all pull requests.
...
-# Create distribution
-python setup.py sdist bdist_wheel
+```bash
+poetry build
 ```
 
+This produces both a `sdist` source distribution, and a `wheel` compiled package.
+
 If the distribution is to be tested locally, then the following command can be used to
 install the source distribution:
```
Comment (davidorme) [inline README.md:114]: "Installed locally"? You'd only want to do this if you were wanting to install it to a python environment *outside* of a development environment managed by `poetry`. Clarify this.

File: README.md [hunk 3]
```diff
@@ -82,17 +125,15 @@ the actual local distribution and **not** the outdated cached version.
 
 ### Publishing a new version
 
-Versions are published to PyPi using `twine`.  As a first step, the new package is
-published to the PyPi **test** site, giving a chance to check the package upload before
-committing it to the live PyPi archive.
+Version publication to PyPi also occurs using `poetry`.  As a first step, the new
+package is published to the PyPi **test** site, giving a chance to check the package
+upload before committing it to the live PyPi archive.
```
Comment (davidorme) [inline README.md:130]: You'll probably need to add notes on configuring credentials for PyPi.

**Review by davidorme — APPROVED** [review]
LGTM

File: README.md (hunk 2 above)
Comment (jacobcook1995) [reply README.md:114]: Ahh yes, this was from the previous README and wasn't sure whether to keep it or not. Probably can just go I guess

File: README.md (hunk 3 above)
Comment (jacobcook1995) [reply README.md:126]: Ahh yes, I also need to learn how to configure PyPi credentials 😅

File: README.md (hunk 1 above)
Comment (jacobcook1995) [reply README.md]: Yeah this plus `premajor/preminor/prepatch` works nicely I feel

File: README.md [hunk 4]
```diff
@@ -52,21 +52,64 @@ whenever commits are made to this repository and for all pull requests.
...
-branch.
+All new package releases should be from the `master` branch, so the changes to `develop`
+have to be moved here. This is achieved using a `release` branch.
+
+```bash
+git branch release/x.y.z
+```
+
+This branch should be pushed to the remote repo to ensure that other developers have
+access to it.
```
[the remark this answers is not in the document]
Comment (jacobcook1995) [reply README.md]: I've added the check list in, I'll work on explaining how to actually add the docs next

**Review by davidorme — COMMENTED** [review]
LGTM - obviously still some details to update.

**Pull request merged**